### ISSUE TITLE
Bug:Lots of `Found an "error" any type` after upgrade to TypeScript 5.5

### ISSUE THREAD

**Issue by stianjensen** [typescript-eslint/typescript-eslint#9736]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

types

### Repro Code

I have a big project, and after updating TypeScript I've noticed that in _some files_, lint rules requiring type information have stopped working. Other eslint rules, and rules from this package that do not require type information still work. When I'm linting a single file inside my editor, all linting also works. Specifying a single file or a single sub-directory on the command line also means linting fully works. But if I ask eslint to lint my entire src/ folder, only some files work.

I've ran with `DEBUG=typescript-eslint:* npx eslint --ext .tsx --ext .ts src --quiet`, and that's when I noticed that every broken file logs one or more instances of `Found an "error" any type`.

I'm not sure where to begin creating a small reproduce of this as I have no idea what part of my project causes only some files to have this issue.

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "extends": "@tsconfig/react-native/tsconfig.json",
  "compilerOptions": {
    "skipLibCheck": true,
    "incremental": true,
    "target": "ES2017",
    "module": "es2015",
    "lib": ["es2019"],
    "jsx": "react-native",
    "sourceMap": true,
    "outDir": "./build",
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "experimentalDecorators": true,
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "useUnknownInCatchVariables": false,
    "resolveJsonModule": true,
    "noUncheckedIndexedAccess": true,
    "isolatedModules": true,
  },
  "include": ["src/**/*"],
}
```

### Expected Result

I would have expected the same lint errors on version TS5.5 of TypeScript as on version TS5.4.

### Actual Result

All lint rules requiring type information (also other packages depending on typescript-eslint, like eslint-deprecation), have stopped working in _some_ files when linting the entire project.

### Additional Info

I'm not sure where to begin debugging further, and have tried looking through more issues here, as well as updating to the 5.6 beta of TypeScript, but the only 'workaround' I've found so far is to downgrade TypeScript again to v5.4

So I'm wondering if there is any other debug output I can try to gather that would be helpful.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     |  I've seen the same issue on 6.x, 7.x and 8.x |
| `TypeScript`                           | `5.5.4` |
| `ESLint`                               | `8.56.0` |
| `node`                                 | `20.10.0` |

**Comment by Josh-Cena** [comment]
It would be impossible for us to investigate if there's no reproduction, and I'm not sure why that TS change would cause type errors.

**Comment by stianjensen** [comment]
> It would be impossible for us (nor for TypeScript) to investigate if there's no reproduction

Yeah, I get that! 😓 

I think I finally managed to track down the source (at least the first important piece)! So I think my original title here was probably mistaken (it was basically the only error I found in the first debug output).

I tracked down that alphabetically every file linted as expected until one specific file, and everything after it was broken. Turns out that the culprit is a file that imports a very large compiled ajv validator (https://ajv.js.org/standalone.html)

I'm not quite sure why this didn't result in an error message or TypeScript outright crashing, but removing the import of that large validator function at least makes the problem go away!
